### DISCUSSION TITLE
Github Actions workflow for Blog plugin not rendering blog?

### DISCUSSION THREAD

**Discussion by rc-sea** [squidfunk/mkdocs-material#6786] [Q&A] [answered]
Hello! Thanks for this wonderful resource. 

I'm trying to launch my personal blog for the first time. I've got everything working properly locally (and it looks fantastic 😎😀🤓 ) but when I try to publish via the Github Actions workflow (using the ci.yml for Insiders documentation: https://squidfunk.github.io/mkdocs-material/publishing-your-site/#with-github-actions-insiders) the site publishes but not the blog? are there libraries I'm not loading? incantantions I'm not uttering?

In addition, when I try to run this snippet i get a file locking error:
`      - run: apt-get install pngquant`

Here's my ci.yml
https://github.com/rc-sea/exuperian/blob/main/.github/workflows/ci.yml

And how the website renders:
https://exuperian.com/

Thanks in advance for your help

rc

**Comment by alexvoss** [comment]
> the site publishes but not the blog?

You mean the blog is empty? A common cause for this is that blog posts are [marked as drafts](https://squidfunk.github.io/mkdocs-material/plugins/blog#meta.draft), so they would render with `mkdocs serve` but not with a `mkdocs build`.

Not sure about the file locking error. Can you share some more info, such as a log?

**Reply by rc-sea** [reply]
sorry I missed your prompt reply. Yes. No blog entry is published at all. I did remove draft status and then the entire tag but the blog still doesn't resolve.

**Reply by rc-sea** [reply]
also i ran a mkdocs build locally and the blog was published in the static site

**Reply by alexvoss** [reply]
Any chance you could produce a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/)? I am happy to have a look to see if I can reproduce the problem and help identify the issue. I don't use Github Pages or Actions so far but have no reason why I could not start now.

**Reply by rc-sea** [reply]
Thank you!  I've attached one here. i'll try to get to a smaller configuration but all of the plug-ins seems to be working locally so don't know why it won't reflect on the Github pages site. I even did a force deploy and that didn't work so it makes me think it's the configuration of the insider's build for linux maybe?  i also wonder if we might need to install these libraries after all (even though the documentation says you don't need to) 
[ubuntu image processing libraries](https://squidfunk.github.io/mkdocs-material/plugins/requirements/image-processing/#cairo-graphics-linux

`apt-get install libcairo2-dev libfreetype6-dev libffi-dev libjpeg-dev libpng-dev libz-dev`

[9.5.9+insiders.4.52.2-min-build.zip](https://github.com/squidfunk/mkdocs-material/files/14301308/9.5.9%2Binsiders.4.52.2-min-build.zip)

**Reply by squidfunk** [reply]
You only need those libraries if you use one of the plugins that do image processing, i.e., the [social](https://squidfunk.github.io/mkdocs-material/plugins/social/) or [optimize](https://squidfunk.github.io/mkdocs-material/plugins/optimize/) plugin. The documentation also mentions that they are already [available when you build on GitHub Actions](https://squidfunk.github.io/mkdocs-material/plugins/requirements/image-processing/#cairo-graphics-linux).

**Reply by rc-sea** [reply]
I'm just trying to throw out ideas for why my blog isn't working or what i can do to address it. i'm at a standstill from publishing standpoint

**Comment by kamilkrzyskow** [comment, marked as answer]
Hello @rc-sea,
as for the error with `apt-get install pngquant` the log says there is an issue with permissions, so I guess the simplest solution is to run the command with `sudo`: https://stackoverflow.com/questions/57982945/how-to-apt-get-install-in-a-github-actions-workflow
`sudo apt-get install -y pngquant`
A more thorough investigation is needed, whether there is a better solution and/or we need to update the documentation.

As for the issue with the lack of Blog on your site, for some reason the deployment isn't being handled/triggered 🤔
You can see here that the last deployment is from 3 days ago https://github.com/rc-sea/exuperian/deployments
Since you save the files (that are used for the deployment) to the `gh-pages` branch, we can check the output of the `mkdocs ...` command.
The https://github.com/rc-sea/exuperian/blob/gh-pages/blog/index.html file (updated yesterday) contains the blog information and is a perfect match when I compare the file contents with a local build. However, if I compare the file with https://exuperian.com/blog/ it's not a match, which just proves that the files weren't updated on the deployment.

I never used CNAMEs or deployed to external domain with GitHub Actions, but the issue lies not in MkDocs or the theme, just some hiccup on your deployment end on GitHub ✌️ 
For now I will suggest, redoing the step to deploy with GitHub Pages to a normal github.io domain, then if that works, switch to your custom domain.

**Reply by rc-sea** [reply]
thank you @kamilkrzyskow ! I'll investigate and report back

**Reply by kamilkrzyskow** [reply]
I see that the https://github.com/rc-sea/exuperian/deployments began to work and https://exuperian.com/blog/ is working now @rc-sea.
However, I don't see any CNAME relevant changes in the commit history, so perhaps you only had to reselect the GitHub Pages deployment in the repository's back-end panel? 

I also saw the addition of the redundant `mkdocs build` https://github.com/rc-sea/exuperian/commit/b8d28fc248ee1824380dc84fb1a9b338201b8013 and removal of the cache https://github.com/rc-sea/exuperian/commit/3985467fff680a392338e6d98fe86e257a4d7011

`mkdocs gh-deploy` will rebuild the site, so if you want to use `mkdocs build` then switch to a deploy action, like in this projects workflow:
https://github.com/squidfunk/mkdocs-material/blob/59f4147e6026ca02d08de2935d9036947e1e4b68/.github/workflows/documentation.yml#L92-L109

I would also like to know what the issue was with caching that forced you to remove it.

For the time being, I'll select my comment as the answer, because there was nothing wrong with the Blog.

**Reply by rc-sea** [reply]
Hi there,

Thanks for the reply. I never got the ci.yml script to actually deploy successfully. I couldn't figure out when updates were occurring so thought it might be a caching issue. To successfully deploy, I ended up using another Github action entirely after committing code and triggering ci.yml. https://github.com/rc-sea/exuperian/actions/workflows/pages/pages-build-deployment

**Reply by kamilkrzyskow** [reply]
> I never got the ci.yml script to actually deploy successfully

As explained in the first comment, the CI.yml workflow worked and deployed the changed files to the `gh-pages` branch. This always worked fine.

> https://github.com/rc-sea/exuperian/actions/workflows/pages/pages-build-deployment 

This is the GitHub internal deployment action, that's how it's supposed to be. It's explained here: https://squidfunk.github.io/mkdocs-material/publishing-your-site/
> If the GitHub Page doesn't show up after a few minutes, go to the settings of
your repository and ensure that the [publishing source branch](https://docs.github.com/en/pages/getting-started-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site) for your GitHub
Page is set to `gh-pages`.

I don't know how it happened that it somehow stopped working after you successfully deployed, but everything seems fine now 👌
